numpy/core/tests/test_internal.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import division, absolute_import, print_function
+
+import ctypes
+import numpy as np
+from numpy.core import _internal
+from numpy.testing import (
+    TestCase, run_module_suite, assert_array_equal, assert_equal)
+
+
+class InternalBaseTest(TestCase):
+    """Base test class for the _internal module."""
+
+    def tearDown(self):
+        # Ensure that ctypes is set on _internal after tests run.
+        _internal.ctypes = ctypes  
```

Comment:
I think this might be better handled as just

```
def test_ctypes_is_not_available(self):
    _internal.ctypes = None
    try:
        do_the_test()
    finally:
        _internal.ctypes = ctypes
```

Or if you want to avoid repetition:

```
def without_ctypes(f):
    @functools.wraps(f)
    def wrapper(*args, **kwargs):
        _internal.ctypes = None
        try:
            return f(*args, **kwargs)
        finally:
            _internal.ctypes = ctypes

@without_ctypes
def test_ctypes_is_not_available(self):
    do_the_test
```

It's a bit confusing as you have it to replace it in contexts where it shouldn't have changed